src/main/java/gov/usgs/wma/waterdata/data/DataController.java

```diff
@@ -0,0 +1,79 @@
+
+import java.io.IOException;
+import java.util.List;
+
+import javax.servlet.http.HttpServletResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.HttpMediaTypeNotAcceptableException;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Observations Data Sets", description = "Data sets such as Statistical Time Series")
+@RestController
+public class DataController extends BaseController {
+	protected TimeSeriesDao timeSeriesDao;
+
+	protected final String contentTypeDesc = "Content format returned, currently only WaterML";
+
+	@Autowired
+	public DataController(TimeSeriesDao timeSeriesDao) {
+		this.timeSeriesDao = timeSeriesDao;
+	}
+
+	@Operation(description = "Return GeoJSON or Water ML Data specific to the requested Monitoring Location and data type.", responses = {
+			@ApiResponse(responseCode = "200", description = "GeoJSON representation of the Time Series.", content = @Content(schema = @Schema(implementation = TimeSeriesGeoJSON.class))),
+			@ApiResponse(responseCode = "400", description = HTTP_400_DESCRIPTION, content = @Content(schema = @Schema(implementation = OgcException.class))),
+			@ApiResponse(responseCode = "404", description = HTTP_404_DESCRIPTION, content = @Content(schema = @Schema(implementation = OgcException.class))),
+			@ApiResponse(responseCode = "500", description = HTTP_500_DESCRIPTION, content = @Content(schema = @Schema(implementation = OgcException.class))) },
+				externalDocs = @ExternalDocumentation(url = "https://github.com/opengeospatial/omsf-profile/tree/master/omsf-json"))
+	@GetMapping(value = "data", produces = { MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE })
+	public void getTimeSeries(
+			@Parameter(description = "Monitoring location Identifier") @RequestParam(value = "monitoringLocationID", required = false) String monLocIdentifier,
+			@Parameter(description = "Data type requested") @RequestParam(value = "type", required = true) DataType dataType,
+			@Parameter(description = "Limits results to time series marked as best = true|false") @RequestParam(value = "best", required = false) Boolean best,
+			@Parameter(description = "Limits data to specfied area") @RequestParam(value = "domain", required = true) List<Domain> domains,
+
+			@Parameter(in = ParameterIn.QUERY, description = contentTypeDesc, schema = @Schema(type = "string"), examples = {
+					@ExampleObject(name = "waterML", value = "WaterML", description = "Water ML") }) @RequestParam(value = "f", required = false, defaultValue = "waterml") String mimeType,
+			HttpServletResponse response) throws HttpMediaTypeNotAcceptableException, IOException {
+
+		determineContentType(mimeType, List.of(ContentType.waterml));
```

Comment:
This is weird, right?  Why do we call a method only for it's side effect?
Instead, this method should just return the ContentType and *not* modify the state of the controller.

Reply:
Good points, in general functions should not have side effects